rest/config.go

```diff
@@ -372,6 +373,13 @@ func (dbConfig *DbConfig) GetCredentials() (string, string, string) {
 	return base.TransformBucketCredentials(dbConfig.Username, dbConfig.Password, *dbConfig.Bucket)
 }
 
+func (dbConfig *DbConfig) ConflictsAllowed() *bool {
+	if dbConfig.AllowConflicts != nil {
+		return dbConfig.AllowConflicts
+	}
+	return base.BooleanPointer(base.DefaultAllowConflicts)
+}
```

Comment:
I agree that it makes more sense to do the nil/default handling here when we parse the config (instead of on each call to databaseContext.AllowConflicts(), but this means that databaseContext.Options.AllowConflicts should just be bool (not *bool), and databaseContext.AllowConflicts() can just return databaseContext.Options.AllowConflicts as-is.

Reply:
@adamcfraser The reason I initially went down this route was that the zero value of a `bool` `databaseContext.Options.AllowConflicts` the would be `false`. Tests which didn't exercise the rest config code (i.e. db package tests) would now have a default of `false` compared to the existing behaviour of defaulting to `true` if unset.

I agree with the comment in general though, and it simplifies the code. I've made the change, and modified tests which require `AllowConflicts: true` to be set.